Error with --about Command on Windows due to Missing sh Command

When running the `--about` command on Windows, the following error is encountered:

```
Caused by: java.io.IOException: Cannot run program "sh": CreateProcess error=2, The system cannot find the file specified
```

This occurs because the current implementation relies on the system's installed Git, which is expected to be available on the user's PATH. On Windows, this can cause issues when Git is not properly configured or when the system tries to invoke the `sh` command, which is typically not available on Windows by default.

#### Background:
Eclipse provides a built-in Git client that is independent of the Git installation on the user's machine. This client is powerful and can be used to avoid relying on an external Git installation, providing better compatibility across different environments.

#### Suggested Improvement:
- **Enhance the Git feature**: Leverage the Git client bundled with Eclipse instead of relying on the Git installed on the user's machine. This would ensure a consistent experience, especially on Windows systems where the `sh` command may not be available.
- **Fallback mechanism**: If Eclipse's Git client is not available, fall back to using the system's installed Git with proper error handling.

#### Steps to Reproduce:
1. Run the `--about` command on a Windows machine where Git is either not installed or improperly configured.
2. Observe the `IOException` related to the missing `sh` command.

#### Expected Behavior:
The command should run successfully without any dependency on an external `sh` or system-installed Git, leveraging Eclipse's built-in Git client.

#### Actual Behavior:
The `IOException` occurs due to the attempt to run the `sh` command, which is not found in the default Windows environment.

#### Suggested Fix:
- Integrate Eclipse's Git client into the application.
- Ensure compatibility with both Windows and other platforms.

--- a/README.md
+++ b/README.md
@@ -56,12 +56,12 @@ Welcome to BuildCLI - Java Project Management!
    mvn package
    ```
 
-3. **Set up BuildCLI for Global Access**:
-    - Copy the `buildcli` file to a directory in your system PATH, such as `~/bin`:
-      ```bash
-      cp target/buildcli ~/bin/buildcli
-      chmod +x ~/bin/buildcli
-      ```
+   3. **Set up BuildCLI for Global Access**:
+       - Copy the `buildcli` file to a directory in your system PATH, such as `~/bin`:
+         ```bash
+         cp target/BuildCLI-1.0-SNAPSHOT.jar ~/bin/buildcli
+         chmod +x ~/bin/buildcli
+         ```
 
 Now `BuildCLI` is ready to use. Test the `buildcli` command in the terminal.
 
@@ -69,32 +69,7 @@ Now `BuildCLI` is ready to use. Test the `buildcli` command in the terminal.
 
 ## Usage
 
-Below are the main commands of `BuildCLI`. For more information about each command, you can run `buildcli --help`.
-
-```plaintext
-Usage: buildcli [OPTIONS]
-Options:
-  -i, --init                Initializes a new Java project
-  -c, --compile             Compiles the Java project
-      --add-dependency      Adds a dependency to the pom.xml in the 'groupId:artifactId' or 'groupId:artifactId:version' format
-      --rm-dependency       Remove a dependency in 'groupId:artifactId' format
-  -p, --profile             Creates a configuration profile (e.g., dev, test)
-  -e, --set-environment     Sets the active profile for the environment (dev, test, prod) by updating `environment.config`
-      --run                 Runs the Java project
-  -d, --document-code       [Beta] Generates documentation for a Java file using AI (e.g., `buildcli -d <path-to-file>.java`)
-  -u, --update              Check for dependency updates
-      --update-now          Update dependencies to latest versions
-  -t, --test                Run tests
-      --dockerize           Generate a Dockerfile for the project
-      --docker-build        Build and run the Docker container
-      --semver              Manage semantic versioning (major, minor, patch)
-      --release             Automate release by creating a Git tag and changelog
-      --cicd-config         Configure CI/CD for the specified tool (e.g., github, gitlab, jenkins)
-  -h, --help                Shows help
-  -V, --version             Shows the version of BuildCLI
-  -a, --about               Displays project information, including its purpose, repository, and contributors.
-      --setup-autocomplete  Set up autocomplete for Bash, Zsh, or Fish
-```
+We made a major refactor of the `BuildCLI` architecture. Please use the `buildcli help` command to see all available options. Also, refer to issue [#89](https://github.com/wheslleyrimar/BuildCLI/issues/89) and pull request [#79](https://github.com/wheslleyrimar/BuildCLI/pull/79) for more details.
 
 ---
 
@@ -107,13 +82,13 @@ You can specify a project name to dynamically set the package structure and proj
 #### Example Commands
 - To initialize a project with a specific name:
 ```bash
-buildcli --init MyProject
+buildcli project init MyProject
 ```
 This will create the project structure with `MyProject` as the base package name, resulting in a directory like `src/main/java/org/myproject`.
 
 - To initialize a project without specifying a name:
 ```bash
-buildcli --init
+buildcli project init
 ```
 This will create the project structure with `buildcli` as the base package name, resulting in a directory like `src/main/java/org/buildcli`.
 
@@ -121,7 +96,7 @@ This will create the project structure with `buildcli` as the base package name,
 ### 2. Compile the Project
 Compiles the Java project using Maven:
 ```bash
-buildcli --compile
+buildcli project build --compile
 ```
 
 ### 3. Add a Dependency to `pom.xml`
@@ -130,37 +105,37 @@ Adds a dependency to the project in the `groupId:artifactId` format. You can als
 #### Example Commands:
 - To add a dependency with the latest version:
 ```bash
-  buildcli --add-dependency org.springframework:spring-core
+  buildcli project add dependency org.springframework:spring-core
 ```
 - To add a dependency with a specified version:
 ```bash
-  buildcli --add-dependency org.springframework:spring-core:5.3.21
+  buildcli p a d org.springframework:spring-core:5.3.21
 ```
 After executing these commands, the dependency will be appended to your pom.xml file under the `<dependencies>` section.
 
 ### 4. Create a Configuration Profile
 Creates a configuration file with the specified profile, for example, `application-dev.properties`:
 ```bash
-buildcli --profile dev
+buildcli project add profile dev
 ```
 
 ### 5. Run the Project
 Runs the Java project using Spring Boot:
 ```bash
-buildcli --run
+buildcli project run
 ```
 
 ### 6. Generate Documentation for Java Code
 Automatically generates inline documentation for a Java file using AI:
 ```bash
-buildcli --document-code File.java
+buildcli project document-code File.java
 ```
 This command sends the specified Java file to the local Ollama server, which generates documentation and comments directly within the code. The modified file with documentation will be saved back to the same location.
 
 ### 7. Set Active Environment Profile
 Sets the active environment profile, saving it to the `environment.config` file. The profile is referenced during project execution, ensuring that the correct configuration is loaded.
 ```bash
-buildcli --set-environment dev
+buildcli p set dev
 ```
 After running this command, the active profile is set to dev, and the `environment.config` file is updated accordingly.
 
@@ -170,27 +145,27 @@ With the `--set-environment` functionality, you can set the active environment p
 ### 8. Dockerize Command
 This command generates a `Dockerfile` for your Java project, making it easier to containerize your application.
 ```bash
-buildcli --dockerize
+buildcli p add dockerfile
 ```
 
 ### 9. Docker Build Command
 This command automatically builds and runs the Docker container for you. After running the command, the Docker image will be created, and your project will run inside the container.
 ```bash
-buildcli --docker-build
+buildcli project build
 ```
 
 ### 10. Set Up CI/CD Integration
 Generates configuration files for CI/CD tools and prepares the project for automated pipelines. Supports Jenkins, Gitlab and Github Actions.
 ```bash
-buildcli --cicd-config github
+buildcli project add pipeline github
 ```
 
 ```bash
-buildcli --cicd-config gitlab
+buildcli project add pipeline gitlab
 ```
 
 ```bash
-buildcli --cicd-config jenkins
+buildcli project add pipeline jenkins
 ```
 
 ---
@@ -198,7 +173,7 @@ buildcli --cicd-config jenkins
 ## Prerequisites
 
 ### Local Ollama API
-Ensure you have the Ollama server running locally, as the `--document-code` functionality relies on an AI model accessible via a local API.
+Ensure you have the Ollama server running locally, as the `docs` functionality relies on an AI model accessible via a local API.
 - [Download Ollama](https://ollama.com/download)
 
 You can start the Ollama server by running:
@@ -216,8 +191,9 @@ ollama run llama3.2
 ## Contribution
 
 Contributions are welcome! Feel free to open **Issues** and submit **Pull Requests**.
+See the [CONTRIBUTING.md](CONTRIBUTING.md) file for more details.
 
-To contribute:
+Quick steps to contribute:
 1. Fork the project.
 2. Create a branch for your changes:
    ```bash

--- a/pom.xml
+++ b/pom.xml
@@ -76,6 +76,12 @@
 			<version>1.5.6</version>
 		</dependency>
 
+		<dependency>
+			<groupId>org.eclipse.jgit</groupId>
+			<artifactId>org.eclipse.jgit</artifactId>
+			<version>6.6.1.202309021850-r</version>
+		</dependency>
+
 	</dependencies>
 
 	<build>
@@ -92,6 +98,16 @@
 							<goal>shade</goal>
 						</goals>
 						<configuration>
+							<filters>
+								<filter>
+									<artifact>*:*</artifact>
+									<excludes>
+										<exclude>META-INF/*.SF</exclude>
+										<exclude>META-INF/*.DSA</exclude>
+										<exclude>META-INF/*.RSA</exclude>
+									</excludes>
+								</filter>
+							</filters>
 							<transformers>
 								<!-- Define a classe principal do JAR -->
 								<transformer

--- a/src/main/java/org/buildcli/domain/git/GitCommandExecutor.java
+++ b/src/main/java/org/buildcli/domain/git/GitCommandExecutor.java
@@ -1,13 +1,24 @@
 package org.buildcli.domain.git;
 
 import org.buildcli.log.SystemOutLogger;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.errors.IncorrectObjectTypeException;
+import org.eclipse.jgit.errors.MissingObjectException;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.transport.URIish;
+import org.slf4j.LoggerFactory;
 
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStreamReader;
+import java.net.URISyntaxException;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import java.util.stream.StreamSupport;
 
 import static org.buildcli.domain.git.GitCommands.*;
 import static org.buildcli.domain.git.GitCommandFormatter.*;
@@ -15,6 +26,71 @@ import static org.buildcli.domain.git.GitCommandFormatter.*;
 public class GitCommandExecutor {
 
     private static final Logger logger = Logger.getLogger(GitCommandExecutor.class.getName());
+    private static final org.slf4j.Logger log = LoggerFactory.getLogger(GitCommandExecutor.class);
+
+    private Git git;
+    private Repository repository;
+
+    private Git openGitRepository(String path) {
+        try {
+            File repoDir = new File(path);
+            git = Git.open(repoDir);
+            return git;
+        } catch (IOException e) {
+            logger.log(Level.SEVERE, "Error opening Git repository", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void setRemoteUrl(String url){
+        try {
+            git.remoteSetUrl().setRemoteUri(new URIish(url));
+        } catch (URISyntaxException e) {
+            logger.log(Level.SEVERE, "Error setting Git remote url", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void gitFetch(){
+        try {
+            git.fetch().call();
+        } catch (GitAPIException e) {
+            logger.log(Level.SEVERE, "Error executing git fetch command", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Repository getRepository(){
+        return git.getRepository();
+    }
+
+    private ObjectId checkRemoteHeadCommits(){
+        try {
+            ObjectId remoteHead = repository.resolve("origin/main^{tree}");
+            if (remoteHead != null){
+                return remoteHead;
+            }
+        } catch (IOException e) {
+            logger.log(Level.SEVERE, "Error resolving remote Head commits");
+        }
+        return null;
+    }
+
+    private ObjectId checkLocalHeadCommits(){
+        try {
+            ObjectId localHead = repository.resolve("HEAD^{tree}");
+            if (localHead != null){
+                return localHead;
+            }
+        } catch (IOException e) {
+            logger.log(Level.SEVERE, "Error resolving local Head commits");
+        }
+        return null;
+    }
+
+    private void closeGitRepository(Git git){
+        git.close();
+    }
 
     protected int runGitCommand(String... command) {
         try {
@@ -57,17 +133,46 @@ public class GitCommandExecutor {
         return output.toString();
     }
 
+    private Iterable<RevCommit> gitLog(){
+        try {
+            return git.log().addPath("src/main/java").call();
+        } catch (GitAPIException e) {
+            logger.log(Level.SEVERE, "Error executing git log command", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Iterable<RevCommit> gitgitLogOnlyCommitsNotInLocal(ObjectId localHead, ObjectId remoteHead){
+        try {
+            return git.log()
+                    .not(localHead)
+                    .add(remoteHead)
+                    .call();
+        } catch (GitAPIException | MissingObjectException | IncorrectObjectTypeException e) {
+            logger.log(Level.SEVERE, "Error executing git log command", e);
+            throw new RuntimeException(e);
+        }
+    }
+
 
 
     public void showContributors(String gitPath) {
-        String localGit = getGitDir(gitPath),
-                workTree = getWorkTree(gitPath);
+        gitPath = "/media/omatheusmesmo/FileShare/buildcli";
+        git = openGitRepository(gitPath);
+        setRemoteUrl("https://github.com/wheslleyrimar/BuildCLI.git");
 
-        String command =String.join(" ",GIT,localGit, workTree, LOG, FORMAT_CONTRIBUTORS, SORT_UNIQUE);
-        String contributors = runGitCommandAndShowOutput("sh", "-c",command);
+        gitFetch();
 
-        SystemOutLogger.log("Contributors:\n");
-        SystemOutLogger.log(contributors);
+        repository = getRepository();
+
+        checkLocalHeadCommits();
+        checkRemoteHeadCommits();
+
+        Iterable<RevCommit> contributors = gitLog();
+
+        SystemOutLogger.log("Contributors: "+ distinctContributors(contributors));
+
+        closeGitRepository(git);
     }
 
     public String getDirectory() {

--- a/src/main/java/org/buildcli/domain/git/GitCommandExecutor.java
+++ b/src/main/java/org/buildcli/domain/git/GitCommandExecutor.java
@@ -157,7 +157,6 @@ public class GitCommandExecutor {
 
 
     public void showContributors(String gitPath) {
-        gitPath = "/media/omatheusmesmo/FileShare/buildcli";
         git = openGitRepository(gitPath);
         setRemoteUrl("https://github.com/wheslleyrimar/BuildCLI.git");
 

--- a/src/main/java/org/buildcli/domain/git/GitCommandFormatter.java
+++ b/src/main/java/org/buildcli/domain/git/GitCommandFormatter.java
@@ -1,5 +1,9 @@
 package org.buildcli.domain.git;
 
+import org.eclipse.jgit.revwalk.RevCommit;
+
+import java.util.stream.StreamSupport;
+
 public class GitCommandFormatter {
 
     protected static String getGitDir(String localRepository){
@@ -12,5 +16,13 @@ public class GitCommandFormatter {
 
     protected static String releaseVersion(String version){
         return "release/v" + version;
+    }
+
+    protected static String distinctContributors(Iterable<RevCommit> contributors){
+        return StreamSupport.stream(contributors.spliterator(), false)
+                .map(revCommit -> revCommit.getAuthorIdent().getName())
+                .distinct()
+                .filter(name -> !name.contains("dependabot"))
+                .toList().toString().replaceAll("[\\[\\]]", "");
     }
 }

